stripecardscan/src/main/java/com/stripe/android/stripecardscan/cardscan/CardScanFragment.kt

```diff
@@ -0,0 +1,306 @@
+package com.stripe.android.stripecardscan.cardscan
+
+import android.annotation.SuppressLint
+import android.content.res.Resources
+import android.graphics.Bitmap
+import android.graphics.PointF
+import android.os.Bundle
+import android.util.Size
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.core.os.bundleOf
+import androidx.core.view.updateMargins
+import androidx.fragment.app.setFragmentResult
+import com.stripe.android.camera.CameraPreviewImage
+import com.stripe.android.camera.framework.Stats
+import com.stripe.android.stripecardscan.R
+import com.stripe.android.stripecardscan.cardscan.exception.InvalidStripePublishableKeyException
+import com.stripe.android.stripecardscan.cardscan.exception.UnknownScanException
+import com.stripe.android.stripecardscan.cardscan.result.MainLoopAggregator
+import com.stripe.android.stripecardscan.cardscan.result.MainLoopState
+import com.stripe.android.stripecardscan.databinding.FragmentCardscanBinding
+import com.stripe.android.stripecardscan.framework.api.dto.ScanStatistics
+import com.stripe.android.stripecardscan.framework.api.uploadScanStats
+import com.stripe.android.stripecardscan.framework.util.AppDetails
+import com.stripe.android.stripecardscan.framework.util.Device
+import com.stripe.android.stripecardscan.payment.card.ScannedCard
+import com.stripe.android.stripecardscan.scanui.CancellationReason
+import com.stripe.android.stripecardscan.scanui.ScanErrorListener
+import com.stripe.android.stripecardscan.scanui.ScanFragment
+import com.stripe.android.stripecardscan.scanui.ScanState
+import com.stripe.android.stripecardscan.scanui.SimpleScanStateful
+import com.stripe.android.stripecardscan.scanui.util.asRect
+import com.stripe.android.stripecardscan.scanui.util.getColorByRes
+import com.stripe.android.stripecardscan.scanui.util.getFloatResource
+import com.stripe.android.stripecardscan.scanui.util.startAnimation
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.launch
+import java.util.concurrent.atomic.AtomicBoolean
+import kotlin.math.min
+import kotlin.math.roundToInt
+
+private val MINIMUM_RESOLUTION = Size(1067, 600) // minimum size of OCR
+const val CARD_SCAN_FRAGMENT_REQUEST_KEY = "CardScanRequestKey"
+const val CARD_SCAN_FRAGMENT_BUNDLE_KEY = "CardScanBundleKey"
+
+class CardScanFragment : ScanFragment(), SimpleScanStateful<CardScanState> {
```

Comment:
can we hide the internal workings of this fragment so that we can keep subclasses and state internal?

Reply:
looks like we can't. Let's just use the `@RestrictTo` annotation where possible.